### ISSUE TITLE
Support for HistGradientBoosting

### ISSUE THREAD

**Issue by ogencoglu** [triton-inference-server/fil_backend#355]
Would be great to have support for scikit-learn's Histogram-based Gradient Boosting which is much simpler and faster than regular gradient boosting.
[Classifier](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.HistGradientBoostingClassifier.html), [Regression](https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.HistGradientBoostingRegressor.html#sklearn.ensemble.HistGradientBoostingRegressor)

**Comment by hcho3** [comment]
@wphicks Treelite added a limited support for HistGradientBoosting (https://github.com/dmlc/treelite/pull/444). The support is not yet complete: `HistGradientBoostingClassifier` with `n_classes > 2` is not supported. To complete this support, we need to release Treelite 4.0 version.

**Comment by hcho3** [comment]
`HistGradientBoostingClassifier` is fully supported in Treelite 4.1. Based on the release schedule, the new Treelite will be incorporate into the FIL backend in release 24.05.

**Comment by hcho3** [comment]
`HistGradientBoostingClassifier` is now fully supported in the latest FIL backend (24.05).
